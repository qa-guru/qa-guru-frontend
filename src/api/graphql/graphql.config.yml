schema: schema.graphql
extensions:
  endpoints:
    Default GraphQL Endpoint:
      url: ${VITE_APP_ENDPOINT}${VITE_GRAPHQL_URI}
      headers:
        user-agent: JS GraphQL
      introspect: false
