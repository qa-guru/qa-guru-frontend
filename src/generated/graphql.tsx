import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
const defaultOptions = {} as const;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  BigDecimal: any;
  BigInteger: any;
  Date: any;
  DateTime: any;
  LocalDateTime: any;
  LocalTime: any;
  Long: any;
  Time: any;
  Timestamp: any;
  Url: any;
  Void: any;
};

export type ContentFileDto = {
  __typename?: 'ContentFileDto';
  fileLocation?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  name?: Maybe<Scalars['String']>;
  size?: Maybe<Scalars['Long']>;
  type?: Maybe<Scalars['String']>;
};

export type HomeWorkDto = {
  __typename?: 'HomeWorkDto';
  contentFiles?: Maybe<Array<Maybe<ContentFileDto>>>;
  description?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  subject?: Maybe<Scalars['String']>;
};

export type HomeWorkInput = {
  description?: InputMaybe<Scalars['String']>;
  id?: InputMaybe<Scalars['ID']>;
  subject?: InputMaybe<Scalars['String']>;
};

export type HomeWorkSort = {
  field?: InputMaybe<HomeWorkSortField>;
  order?: InputMaybe<Order>;
};

export enum HomeWorkSortField {
  CreationDate = 'CREATION_DATE',
  Subject = 'SUBJECT'
}

export type HomeWorksDto = {
  __typename?: 'HomeWorksDto';
  items?: Maybe<Array<Maybe<HomeWorkDto>>>;
  totalElements?: Maybe<Scalars['Long']>;
  totalPages?: Maybe<Scalars['Int']>;
};

/** Mutation root */
export type Mutation = {
  __typename?: 'Mutation';
  /** purchase section */
  createPurchase?: Maybe<PurchaseDto>;
  /** training section */
  createTraining?: Maybe<TrainingDto>;
  /** user section */
  createUser?: Maybe<UserDto>;
  deleteHomeWork?: Maybe<Scalars['Void']>;
  deleteTask?: Maybe<Scalars['Void']>;
  deleteUser?: Maybe<Scalars['Void']>;
  /** homeWork section */
  updateHomeWork?: Maybe<HomeWorkDto>;
  /** person section */
  updatePerson?: Maybe<PersonDto>;
  updateRole?: Maybe<UserDto>;
  /** task section */
  updateTask?: Maybe<TaskDto>;
};


/** Mutation root */
export type MutationCreatePurchaseArgs = {
  input: PurchaseInput;
};


/** Mutation root */
export type MutationCreateTrainingArgs = {
  input: TrainingInput;
};


/** Mutation root */
export type MutationCreateUserArgs = {
  password: Scalars['String'];
  username: Scalars['String'];
};


/** Mutation root */
export type MutationDeleteHomeWorkArgs = {
  id: Scalars['ID'];
};


/** Mutation root */
export type MutationDeleteTaskArgs = {
  id: Scalars['ID'];
};


/** Mutation root */
export type MutationDeleteUserArgs = {
  id: Scalars['ID'];
};


/** Mutation root */
export type MutationUpdateHomeWorkArgs = {
  input: HomeWorkInput;
};


/** Mutation root */
export type MutationUpdatePersonArgs = {
  input: PersonInput;
};


/** Mutation root */
export type MutationUpdateRoleArgs = {
  id: Scalars['ID'];
  roles?: InputMaybe<Array<InputMaybe<UserRole>>>;
};


/** Mutation root */
export type MutationUpdateTaskArgs = {
  input: TaskInput;
};

export enum Order {
  Asc = 'ASC',
  Desc = 'DESC'
}

export type PersonDto = {
  __typename?: 'PersonDto';
  avatarLocation?: Maybe<Scalars['String']>;
  firstName?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  lastName?: Maybe<Scalars['String']>;
  middleName?: Maybe<Scalars['String']>;
  phoneNumber?: Maybe<Scalars['String']>;
};

export type PersonInput = {
  firstName: Scalars['String'];
  lastName: Scalars['String'];
  middleName?: InputMaybe<Scalars['String']>;
  phoneNumber: Scalars['String'];
};

export type PurchaseDto = {
  __typename?: 'PurchaseDto';
  id?: Maybe<Scalars['ID']>;
  training: TrainingDto;
  trainingTariff: TrainingTariffDto;
  user: UserDto;
};

export type PurchaseInput = {
  trainingId?: InputMaybe<Scalars['ID']>;
  trainingTariffId?: InputMaybe<Scalars['ID']>;
  userId?: InputMaybe<Scalars['ID']>;
};

/** Query root */
export type Query = {
  __typename?: 'Query';
  /** homeWork section */
  homeWork?: Maybe<HomeWorkDto>;
  homeWorks?: Maybe<HomeWorksDto>;
  /** person section */
  person?: Maybe<PersonDto>;
  personByUserId?: Maybe<PersonDto>;
  /** task section */
  task?: Maybe<TaskDto>;
  taskHomeWorks?: Maybe<Array<Maybe<HomeWorkDto>>>;
  tasks?: Maybe<TasksDto>;
  /** user section */
  user?: Maybe<UserDto>;
  users?: Maybe<UsersDto>;
};


/** Query root */
export type QueryHomeWorkArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryHomeWorksArgs = {
  page: Scalars['Int'];
  size: Scalars['Int'];
  sort?: InputMaybe<HomeWorkSort>;
};


/** Query root */
export type QueryPersonByUserIdArgs = {
  id: Scalars['ID'];
};


/** Query root */
export type QueryTaskArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryTaskHomeWorksArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryTasksArgs = {
  page: Scalars['Int'];
  size: Scalars['Int'];
  sort?: InputMaybe<TaskSort>;
};


/** Query root */
export type QueryUserArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryUsersArgs = {
  page: Scalars['Int'];
  size: Scalars['Int'];
  sort?: InputMaybe<UserSort>;
};

export type TaskDto = {
  __typename?: 'TaskDto';
  contentFiles?: Maybe<Array<Maybe<ContentFileDto>>>;
  description?: Maybe<Scalars['String']>;
  homeWorks?: Maybe<Array<Maybe<HomeWorkDto>>>;
  id?: Maybe<Scalars['ID']>;
  last?: Maybe<Scalars['ID']>;
  locking?: Maybe<Scalars['Boolean']>;
  speakers?: Maybe<Array<Maybe<UserInfoDto>>>;
  subject?: Maybe<Scalars['String']>;
};

export type TaskInput = {
  description?: InputMaybe<Scalars['String']>;
  homeWorks?: InputMaybe<Array<InputMaybe<Scalars['String']>>>;
  id?: InputMaybe<Scalars['ID']>;
  last?: InputMaybe<Scalars['ID']>;
  locking?: InputMaybe<Scalars['Boolean']>;
  speakers?: InputMaybe<Array<InputMaybe<Scalars['String']>>>;
  subject?: InputMaybe<Scalars['String']>;
};

export type TaskSort = {
  field?: InputMaybe<TaskSortField>;
  order?: InputMaybe<Order>;
};

export enum TaskSortField {
  CreationDate = 'CREATION_DATE',
  Subject = 'SUBJECT'
}

export type TasksDto = {
  __typename?: 'TasksDto';
  items?: Maybe<Array<Maybe<TaskDto>>>;
  totalElements?: Maybe<Scalars['Long']>;
  totalPages?: Maybe<Scalars['Int']>;
};

export enum TechStack {
  Java = 'JAVA',
  Python = 'PYTHON'
}

export type TrainingDto = {
  __typename?: 'TrainingDto';
  id: Scalars['ID'];
  name: Scalars['String'];
  tariffs?: Maybe<Array<Maybe<TrainingTariffDto>>>;
  techStack: TechStack;
};

export type TrainingInput = {
  name?: InputMaybe<Scalars['String']>;
  tariffs?: InputMaybe<Array<InputMaybe<TrainingTariffInput>>>;
  techStack: TechStack;
};

export type TrainingTariffDto = {
  __typename?: 'TrainingTariffDto';
  code?: Maybe<Scalars['String']>;
  description?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  name?: Maybe<Scalars['String']>;
};

export type TrainingTariffInput = {
  code?: InputMaybe<Scalars['String']>;
  description?: InputMaybe<Scalars['String']>;
  name?: InputMaybe<Scalars['String']>;
};

export type UserDto = {
  __typename?: 'UserDto';
  email?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  roles?: Maybe<Array<Maybe<UserRole>>>;
};

export type UserIdInput = {
  id?: InputMaybe<Scalars['ID']>;
};

export type UserInfoDto = {
  __typename?: 'UserInfoDto';
  email?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  person?: Maybe<PersonDto>;
  roles?: Maybe<Array<Maybe<UserRole>>>;
};

export enum UserRole {
  Admin = 'ADMIN',
  Lector = 'LECTOR',
  Manager = 'MANAGER',
  Master = 'MASTER',
  Mentor = 'MENTOR',
  User = 'USER'
}

export type UserSort = {
  field?: InputMaybe<UserSortField>;
  order?: InputMaybe<Order>;
};

export enum UserSortField {
  Email = 'EMAIL',
  LastName = 'LAST_NAME',
  Phone = 'PHONE'
}

export type UsersDto = {
  __typename?: 'UsersDto';
  items?: Maybe<Array<Maybe<UserInfoDto>>>;
  totalElements?: Maybe<Scalars['Long']>;
  totalPages?: Maybe<Scalars['Int']>;
};

export type SignUpMutationVariables = Exact<{
  username: Scalars['String'];
  password: Scalars['String'];
}>;


export type SignUpMutation = { __typename?: 'Mutation', createUser?: { __typename?: 'UserDto', id?: string | null, email?: string | null, roles?: Array<UserRole | null> | null } | null };


export const SignUpDocument = gql`
    mutation SignUp($username: String!, $password: String!) {
  createUser(username: $username, password: $password) {
    id
    email
    roles
  }
}
    `;
export type SignUpMutationFn = Apollo.MutationFunction<SignUpMutation, SignUpMutationVariables>;

/**
 * __useSignUpMutation__
 *
 * To run a mutation, you first call `useSignUpMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useSignUpMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [signUpMutation, { data, loading, error }] = useSignUpMutation({
 *   variables: {
 *      username: // value for 'username'
 *      password: // value for 'password'
 *   },
 * });
 */
export function useSignUpMutation(baseOptions?: Apollo.MutationHookOptions<SignUpMutation, SignUpMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<SignUpMutation, SignUpMutationVariables>(SignUpDocument, options);
      }
export type SignUpMutationHookResult = ReturnType<typeof useSignUpMutation>;
export type SignUpMutationResult = Apollo.MutationResult<SignUpMutation>;
export type SignUpMutationOptions = Apollo.BaseMutationOptions<SignUpMutation, SignUpMutationVariables>;