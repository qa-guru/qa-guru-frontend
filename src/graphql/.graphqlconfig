{
  "name": "Untitled GraphQL Schema",
  "schemaPath": "schema.graphql",
  "extensions": {
    "endpoints": {
      "Default GraphQL Endpoint": {
        "url": "http://5.161.103.109:8080/graphql",
        "headers": {
          "user-agent": "JS GraphQL"
        },
        "introspect": false
      }
    }
  }
}