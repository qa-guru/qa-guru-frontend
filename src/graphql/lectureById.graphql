query LectureById($id: ID) {
  lecture(id: $id) {
    id
    subject
    description
    speakers {
      id
      email
      roles
      person {
        firstName
        lastName
        middleName
        phoneNumber
        avatarLocation
      }
    }
    lectureHomeWorks {
      id
      subject
      description
    }
  }
}
