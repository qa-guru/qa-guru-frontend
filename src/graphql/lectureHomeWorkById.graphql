query LectureHomeWorkById($id: ID) {
  lectureHomeWork(id: $id) {
    id
    subject
    description
  }
}
