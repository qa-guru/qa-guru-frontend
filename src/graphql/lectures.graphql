query Lectures {
  lectures(page: 0, size: 10, sort: { field: SUBJECT }) {
    items {
      id
      subject
      description
      speakers {
        id
        email
        roles
        person {
          firstName
          lastName
          middleName
          phoneNumber
          avatarLocation
        }
      }
      lectureHomeWorks {
        id
        subject
        description
      }
    }
    totalPages
    totalElements
  }
}
