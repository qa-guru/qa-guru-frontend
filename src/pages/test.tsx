import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
