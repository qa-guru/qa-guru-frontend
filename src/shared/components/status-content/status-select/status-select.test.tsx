import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "test/utils-test";
import { StudentHomeWorkStatus } from "api/graphql/generated/graphql";

import StatusSelect from "./status-select";

describe("StatusSelect", () => {
  const homeworkId = "homework-id";

  it("status is NEW and user is the current mentor", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <StatusSelect
          currentStatus={StudentHomeWorkStatus.New}
          homeworkId={homeworkId}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("status is IN_REVIEW and user is the current mentor", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <StatusSelect
          currentStatus={StudentHomeWorkStatus.InReview}
          homeworkId={homeworkId}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("status is APPROVED and user is not the current mentor", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <StatusSelect
          currentStatus={StudentHomeWorkStatus.Approved}
          homeworkId={homeworkId}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("status is NOT_APPROVED and user is the current mentor", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <StatusSelect
          currentStatus={StudentHomeWorkStatus.NotApproved}
          homeworkId={homeworkId}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
