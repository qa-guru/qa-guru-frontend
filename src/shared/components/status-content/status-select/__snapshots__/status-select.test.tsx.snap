// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatusSelect > status is APPROVED and user is not the current mentor 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1j82rsr-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        id="status-select-label"
      >
        Статус
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl  css-d4zdg1-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-disabled="true"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="status-select"
          class="MuiSelect-select MuiSelect-outlined Mui-disabled MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1c0lsac-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="status-select"
          role="button"
        >
          <div
            class="MuiStack-root css-19iseno-MuiStack-root"
          >
            <svg
              class="css-8a7nt6"
              fill="none"
              height="14"
              viewBox="0 0 18 14"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.00016 11.17L1.83016 7L0.410156 8.41L6.00016 14L18.0002 2L16.5902 0.59L6.00016 11.17Z"
                fill="#2CCCA6"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-158lv7p-MuiTypography-root"
            >
              Принято
            </p>
          </div>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          disabled=""
          tabindex="-1"
          value="APPROVED"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined Mui-disabled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Статус
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusSelect > status is IN_REVIEW and user is the current mentor 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1j82rsr-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        id="status-select-label"
      >
        Статус
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-d4zdg1-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="status-select"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-1c0lsac-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="status-select"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiStack-root css-19iseno-MuiStack-root"
          >
            <svg
              class="css-8a7nt6"
              fill="none"
              height="18"
              viewBox="0 0 18 18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.76 10.27L17.49 16L16 17.49L10.27 11.76C9.2 12.53 7.91 13 6.5 13C2.91 13 0 10.09 0 6.5C0 2.91 2.91 0 6.5 0C10.09 0 13 2.91 13 6.5C13 7.91 12.53 9.2 11.76 10.27ZM6.5 2C4.01 2 2 4.01 2 6.5C2 8.99 4.01 11 6.5 11C8.99 11 11 8.99 11 6.5C11 4.01 8.99 2 6.5 2Z"
                fill="#6750A4"
                fill-rule="evenodd"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-158lv7p-MuiTypography-root"
            >
              На проверке
            </p>
          </div>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="IN_REVIEW"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Статус
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusSelect > status is NEW and user is the current mentor 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1j82rsr-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        id="status-select-label"
      >
        Статус
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-d4zdg1-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="status-select"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-1c0lsac-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="status-select"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiStack-root css-19iseno-MuiStack-root"
          >
            <svg
              class="css-8a7nt6"
              fill="none"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.99 0C4.47 0 0 4.48 0 10C0 15.52 4.47 20 9.99 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 9.99 0ZM10 18C5.58 18 2 14.42 2 10C2 5.58 5.58 2 10 2C14.42 2 18 5.58 18 10C18 14.42 14.42 18 10 18Z"
                fill="black"
                fill-opacity="0.56"
              />
              <path
                d="M10.5 5H9V11L14.25 14.15L15 12.92L10.5 10.25V5Z"
                fill="black"
                fill-opacity="0.56"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-158lv7p-MuiTypography-root"
            >
              Ожидает проверки
            </p>
          </div>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="NEW"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Статус
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusSelect > status is NOT_APPROVED and user is the current mentor 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1j82rsr-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        id="status-select-label"
      >
        Статус
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-d4zdg1-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="status-select"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-1c0lsac-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="status-select"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiStack-root css-19iseno-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-16jd6n3-MuiSvgIcon-root"
              data-testid="HighlightOffIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14.59 8 12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-158lv7p-MuiTypography-root"
            >
              Не принято
            </p>
          </div>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="NOT_APPROVED"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Статус
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
